src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -356,6 +358,45 @@ def list_available_extensions(index_url=None, show_details=False):
     return results
 
 
+def list_versions(extension_name, index_url=None):
+    index_data = get_index_extensions(index_url=index_url)
+
+    try:
+        exts = index_data[extension_name]
+    except Exception:
+        raise CLIError('Extension {} not found.'.format(extension_name))
```

Comment:
should this error out or silently swallow? will this func be used other than az upgrade?

Reply:
This function is not used in `az upgrade` now as we always update to the latest extension version. This error happens when users type the wrong extension name.

Reply:
ok. I saw it used by list-available-version, so error out is ok. My original point is that az upgrade internal failure should not error out which might impact user in silent upgrading scenario.

Reply:
In silent upgrading scenario (auto-upgrade with no prompt), the exit code and `stdout`of a command will not be affected, but warnings and errors from `az upgrade` will be shown in `stderr`.